presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ApplyNode.java

```diff
@@ -62,19 +63,26 @@
      */
     private final Assignments subqueryAssignments;
 
+    /**
+     * Used for error reporting in case this ApplyNode is not supported
+     */
+    private final Node originSubquery;
+
     @JsonCreator
     public ApplyNode(
             @JsonProperty("id") PlanNodeId id,
             @JsonProperty("input") PlanNode input,
             @JsonProperty("subquery") PlanNode subquery,
             @JsonProperty("subqueryAssignments") Assignments subqueryAssignments,
-            @JsonProperty("correlation") List<Symbol> correlation)
+            @JsonProperty("correlation") List<Symbol> correlation,
+            @JsonProperty("originSubquery") Node originSubquery)
```

Comment:
There is also a predicate in `TableScan`. We send `Expressions` over the wire with same, and also whole query is serialized and displayed within the webui.